### DISCUSSION TITLE
Linking board to HAL (NXP).

### DISCUSSION THREAD

**Discussion by JaagupAverin** [zephyrproject-rtos/zephyr#49157] [Q&A] [answered]
Hello. Could someone help me understand the relationship between a board/dts/soc and HAL module.

I am using IMX8MM board as an example but cannot understand how it will link to the NXP HAL.
It is apparent from the NXP HAL CMake that it only does anything if HAS_MCUX or HAS_IMX_HAL is defined. (https://github.com/zephyrproject-rtos/hal_nxp/blob/master/CMakeLists.txt)

The board or its related files define neither: https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/arm64/imx8mm_evk/doc/index.rst

Where are the symbols located that define the relationship between a particular board and the HAL? Thank you.

**Comment by JiafeiPan** [comment, marked as answer]
Current IMX8MM board support in Zephyr community is not using NXP HAL, but NXP is upstreaming some more flexible board support with more features, includes IMX8MM, IMX8MP and IMX8MN, they will enable HAS_MCUX and will leverage some drivers in NXP HAL.

**Reply by pefech** [reply]
Hi Jiafei, we at PHYTEC would really like to switch from MCUX to Zephyr as real-time operating system on our i.MX8MM and i.MX8MP boards because of it's much benefits. To plan our developement it would be really nice, if you know when NXP plans to send the new HAL drivers for the i.MX8MM and i.MX8MP upstream. Thank you!

**Reply by JiafeiPan** [reply]
Hi, prefech, we will upstream new hal driver ASAP, and hope it will be around the end of this month. Hope the plan is okay for you, and let me know if you have more comments, thanks.

**Reply by pefech** [reply]
Hi, that fits perfect! Thank you very much for your answer!

**Reply by JiafeiPan** [reply]
FYI, I have upstreamed new platfomr imx8mn, and also updated imx8mm and imxmp, please refer to the following PR and it is in review progress:

https://github.com/zephyrproject-rtos/zephyr/pull/49651

**Reply by pefech** [reply]
thanks!

**Comment by wqiiqw** [comment]
Dear all, I have the same question. I can't find the exact "string" to match board to its HAL. Any hints will be appreciated. Thank you.

BTW, I am adding a new SOC (ARM cortext M4) with many peripherals and I need to create the "new_vendor" HAL for the platform init codes.

**Reply by DerekSnell** [reply]
Hi @wqiiqw ,
This is an old outdated discussion, and is specific to the NXP HAL.  It seems you are attempting to create a new HAL, and your question is unrelated to this discussion.

I suggest you create a new Discussion to get more attention from the Zephyr Community.  Best regards
